### PR TITLE
doc: HWMv2: align board terminology

### PR DESCRIPTION
**Pull request by greg-fer**
Changed `build target` to `board target` as per HWMv2 terminology.
Updated leftover v1 board target names.
Updated glossary.
Updated board names page with some additional info and links to Zephyr.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

----

Related Zephyr docs used as reference: https://docs.nordicsemi.com/bundle/ncs-latest/page/zephyr/hardware/porting/board_porting.html#board_terminology

### CODE REVIEW

File: doc/nrf/glossary.rst [hunk 1]
```diff
@@ -182,6 +184,10 @@ Glossary
 
    Development Kit (DK)
       A hardware development platform used for application development.
+      See :ref:`app_boards_names` for more information.
+
+   Device
+      Any piece of hardware that is compatible with the |NCS|, such as a :term:`development kit (DK)` provided by Nordic Semiconductor, prototyping platforms such as Nordic Thingy:53 or Nordic Thingy:91, or third-party development kits compatible with Nordic :term:`boards`.
```
Comment (divipillai) [inline doc/nrf/glossary.rst:190]:
```suggestion
      Any piece of hardware that is compatible with the |NCS|, such as a :term:`development kit (DK)` provided by Nordic Semiconductor, prototyping platforms such as Nordic Thingy:53 or Nordic Thingy:91, or third-party development kits compatible with Nordic :term:`boards <Board>`.
```

File: doc/nrf/glossary.rst [hunk 2]
```diff
@@ -79,9 +79,11 @@ Glossary
       A layer of the Bluetooth LE protocol stack comprised of multiple (non real-time) network and transport protocols enabling applications to communicate with peer devices in a standard and interoperable way.
 
    Board
-      In Zephyr and the |NCS|, the name of a specific supported hardware target.
+      In Zephyr and the |NCS|, the name of a specific supported hardware target that can be part of a :term:`device`.
```
Comment (divipillai) [inline doc/nrf/glossary.rst:82]:
```suggestion
      In Zephyr and the |NCS|, the name of a specific supported hardware target that can be part of a :term:`device <Device>`.
```

**Review by divipillai — APPROVED** [review]

**Review by edmont — APPROVED** [review]
Zigbee ✔️

File: doc/nrf/device_guides/working_with_nrf/nrf91/nrf91_features.rst [hunk 3]
```diff
@@ -43,7 +43,7 @@ Trusted Firmware-M (TF-M)
 
 Trusted Firmware-M provides a configurable set of software components to create a Trusted Execution Environment.
 It has replaced Secure Partition Manager as the solution used by |NCS| applications and samples.
-This means that when you build your application for ``_ns`` build targets, TF-M is automatically included in the build.
+This means that when you build your application for ``_ns`` board targets, TF-M is automatically included in the build.
```
Comment (gmarull) [inline doc/nrf/device_guides/working_with_nrf/nrf91/nrf91_features.rst:46]:
```suggestion
This means that when you build your application for ``*/ns`` board targets, TF-M is automatically included in the build.
```

File: doc/nrf/device_guides/working_with_nrf/nrf91/thingy91.rst [hunk 4]
```diff
@@ -520,26 +520,26 @@ Building and programming from the source code
...
 
-   * In |NCS| releases before v1.3.0, these build targets were named ``nrf9160_pca20035``, ``nrf9160_pca20035ns``, and ``nrf52840_pca20035``.
-   * In |NCS| releases ranging from v1.3.0 to v1.6.1, the build target ``thingy91/nrf9160/ns`` was named ``thingy91_nrf9160ns``.
+   * In |NCS| releases before v1.3.0, these board targets were named ``nrf9160_pca20035``, ``nrf9160_pca20035ns``, and ``nrf52840_pca20035``.
+   * In |NCS| releases ranging from v1.3.0 to v1.6.1, the board target ``thingy91/nrf9160/ns`` was named ``thingy91_nrf9160ns``.
 
 .. note::
 
-   LTE/GNSS features can only be used with :ref:`Cortex-M Security Extensions enabled <app_boards_spe_nspe_cpuapp_ns>` (``_ns`` build target).
+   LTE/GNSS features can only be used with :ref:`Cortex-M Security Extensions enabled <app_boards_spe_nspe_cpuapp_ns>` (``_ns`` board target).
```
Comment (gmarull) [inline doc/nrf/device_guides/working_with_nrf/nrf91/thingy91.rst:542]:
```suggestion
   LTE/GNSS features can only be used with :ref:`Cortex-M Security Extensions enabled <app_boards_spe_nspe_cpuapp_ns>` (``/ns`` board target).
```

File: doc/nrf/releases_and_maturity/releases/release-notes-1.8.0.rst [hunk 5]
```diff
@@ -195,7 +195,7 @@ nRF Desktop
 
   * Added documentation for :ref:`nrf_desktop_usb_state_pm`.
   * Added :ref:`nrf_desktop_ble_state_pm`.
-  * Added peer erase feature for the ``nrf52840dk_nrf52840`` build target.
+  * Added peer erase feature for the ``nrf52840dk_nrf52840`` board target.
```
Comment (gmarull) [inline doc/nrf/releases_and_maturity/releases/release-notes-1.8.0.rst:198]: I think we should not change terminology in release notes

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 6]
```diff
@@ -257,7 +257,7 @@ nRF Desktop
     The value is now aligned with the Fast Pair requirements.
   * Enabled the :ref:`CONFIG_DESKTOP_CONFIG_CHANNEL_OUT_REPORT <config_desktop_app_options>` Kconfig option for the nRF Desktop peripherals with :ref:`nrf_desktop_dfu`.
     The option mitigates HID report rate drops during DFU image transfer through the nRF Desktop dongle.
-    The output report is also enabled for the ``nrf52kbd_nrf52832`` build target in the debug configuration to maintain consistency with the release configuration.
+    The output report is also enabled for the ``nrf52kbd_nrf52832`` board target in the debug configuration to maintain consistency with the release configuration.
```
Comment (gmarull) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:260]:
```suggestion
    The output report is also enabled for the ``nrf52kbd/nrf52832`` board target in the debug configuration to maintain consistency with the release configuration.
```

(are these latest rel notes?)

File: samples/matter/light_switch/README.rst [hunk 7]
```diff
@@ -43,8 +43,8 @@ IPv6 network support
 
 The development kits for this sample offer the following IPv6 network support for Matter:
 
-* Matter over Thread is supported for ``nrf52840dk_nrf52840``, ``nrf5340dk_nrf5340_cpuapp``, ``nrf21540dk_nrf52840``, and ``nrf54l15pdk_nrf54l15``.
```
Comment (kkasperczyk-no) [inline samples/matter/light_switch/README.rst:46]: I noticed  there should be `nrf54l15pdk_nrf54l15_cpuapp` (it was wrong before your change). Could you fix this btw of making some next push? The same for the other Matter samples.

**Review by kkasperczyk-no — APPROVED** [review]
Looks good from the Matter side.

File: doc/nrf/config_and_build/board_support/board_names.rst [hunk 8]
```diff
@@ -8,23 +8,39 @@ Board names
...
+* :term:`System on Chip (SoC)` - When you choose a board target with this qualifier, you build for this specific SoC on this board.
+
+* :ref:`CPU cluster <zephyr:glossary>` - You can use this board qualifier to build for a group of one or more CPU cores, all executing the same image within the same address space and in a symmetrical (SMP) configuration.
+  The CPU cluster board qualifier varies depending on the SoC and SoC Series.
+  For example, :ref:`building <building>` for the ``thingy53/nrf5340/cpunet`` board target builds the network core firmware for Nordic Thingy:53, building for ``nrf54h20dk/nrf54h20/cpuapp`` builds the application core firmware for the nRF54H20 DK, building for ``nrf54h20dk/nrf54h20/cpurad`` builds the radio core firmware for the nRF54H20 DK, and so on.
+  Check the Product Specification of the given SoC for more information about the available CPU clusters.
+
+* :ref:`Variant <zephyr:glossary>` - You can use this board qualifier to build for a particular type or configuration of a build for a combination of SoC and CPU cluster.
```
Comment (wiba-nordic) [inline doc/nrf/config_and_build/board_support/board_names.rst:35]: Should variant be in the table?

File: doc/nrf/config_and_build/board_support/board_names.rst [hunk 9]
```diff
@@ -40,7 +56,7 @@ Also see the :ref:`zephyr:boards` section in the Zephyr documentation.
 .. _table:
 
 +-------------------+------------+-------------------------------------------------------------------+---------------------------------------+
-| Hardware platform | PCA number | Board name                                                        | Build target                          |
+| Hardware platform | PCA number | Board name                                                        | Board targets                          |
```
Comment (wiba-nordic) [inline doc/nrf/config_and_build/board_support/board_names.rst:59]:
```suggestion
| Hardware platform | PCA number | Board name                                                        | Board targets                         |
```

File: doc/nrf/config_and_build/board_support/board_names.rst [hunk 10]
```diff
@@ -119,7 +135,7 @@ Boards included in sdk-nrf
 The following boards are defined in the :file:`nrf/boards/nordic/` folder.
 
 +-------------------+------------+----------------------------------------------------------+---------------------------------------+
-| Hardware platform | PCA number | Board name                                               | Build target                          |
+| Hardware platform | PCA number | Board name                                               | Board targets                          |
```
Comment (wiba-nordic) [inline doc/nrf/config_and_build/board_support/board_names.rst:138]:
```suggestion
| Hardware platform | PCA number | Board name                                               | Board targets                         |
```

File: doc/nrf/config_and_build/board_support/board_names.rst [hunk 11]
```diff
@@ -160,7 +176,7 @@ Shields included in sdk-nrf
 The following shields are defined in the :file:`nrf/boards/shields` folder.
 
 +----------------------------------------------------------+------------+--------------------------------------------------------------------------+---------------------------------------+
-| Hardware platform                                        | PCA number | Board name                                                               | Build target                          |
+| Hardware platform                                        | PCA number | Board name                                                               | Board targets                          |
```
Comment (wiba-nordic) [inline doc/nrf/config_and_build/board_support/board_names.rst:179]:
```suggestion
| Hardware platform                                        | PCA number | Board name                                                              | Board targets                         |
```

File: doc/nrf/config_and_build/board_support/index.rst [hunk 12]
```diff
@@ -3,10 +3,12 @@
 Board support
 #############
 
-The |NCS| provides board definitions for all Nordic Semiconductor devices.
-These are in tables on the :ref:`app_boards_names` page, listed by :ref:`app_boards_names_zephyr`, :ref:`app_boards_names_nrf`, and :ref:`shield_names_nrf`.
+The |NCS| provides board definitions for all Nordic Semiconductor devices and follows Zephyr's :ref:`hw_support_hierarchy`.
```
Comment (wiba-nordic) [inline doc/nrf/config_and_build/board_support/index.rst:6]:
```suggestion
The |NCS| provides board definitions for all Nordic Semiconductor devices and follows Zephyr's :ref:`zephyr:hw_support_hierarchy`.
```

File: doc/nrf/device_guides/working_with_nrf/nrf91/nrf91_features.rst (hunk 3 above)
Comment (wiba-nordic) [reply doc/nrf/device_guides/working_with_nrf/nrf91/nrf91_features.rst:46]: I think this was labeled a variant on the Board names page now. Should that terminology be used here as well?

File: doc/nrf/releases_and_maturity/releases/release-notes-1.8.0.rst (hunk 5 above)
Comment (wiba-nordic) [reply doc/nrf/releases_and_maturity/releases/release-notes-1.8.0.rst:198]: Agreed.

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 6 above)
Comment (greg-fer) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:260]: These are latest rel notes, not yet released. I'll fix.

**Review by tejlmand — COMMENTED** [review]
Please use `shield` instead of shield board target.

Also we do have a term for when the board target name is used for files, and that is `normalized board target`.

Rest looks good 👍

**Review by kkasperczyk-no — APPROVED** [review]

File: doc/nrf/config_and_build/board_support/board_names.rst [hunk 13]
```diff
@@ -8,23 +8,39 @@ Board names
...
++=============+================+=========================+=================================+
+| Board name  | Board revision | Board qualifier for SoC | Board qualifier for CPU cluster |
++-------------+----------------+-------------------------+---------------------------------+
 
-  * ``cpuapp`` - When you choose this target, you build the application core firmware.
-  * ``cpunet`` - When you choose this target, you build the network core firmware.
+While the board name is always present, other elements, such as the board revision or the board qualifiers, are optional:
 
-* For usage of Cortex-M Security Extensions (CMSE):
+* :ref:`Board revision <zephyr:glossary>` - You can use the *board_name@board_revision* board target to get extra devicetree overlays with new features available for a specific board version.
```
[the remark this answers is not in the document]
Comment (greg-fer) [reply doc/nrf/config_and_build/board_support/board_names.rst:24]: About linking to Zephyr vs including the terms in the glossary: I'm worried of getting inconsistencies if we duplicate these entries. The Board entry in our glossary links to Zephyr as a middle-ground solution...

File: doc/nrf/config_and_build/board_support/board_names.rst (hunk 8 above)
Comment (greg-fer) [reply doc/nrf/config_and_build/board_support/board_names.rst:35]: There's no viable example. But I've added an empty field.

**Comment by greg-fer** [comment]
@wiba-nordic , applied your edits. For some reason, I completely missed most of them :/

**Review by plskeggs — APPROVED** [review]

**Review by glarsennordic — APPROVED** [review]

**Review by wiba-nordic — APPROVED** [review]

**Review by SeppoTakalo — APPROVED** [review]

**Pull request merged**